crates/driver/src/settlement_proposal.rs

```diff
@@ -1,464 +0,0 @@
-        verify_executed_amount(&self.order, self.executed_amount)?;
-        let remaining = shared::remaining_amounts::Remaining::from_order(&self.order)?;
-        let remaining_sell = remaining
-            .remaining(self.order.data.sell_amount)
-            .context("remaining sell")?;
-        let remaining_buy = remaining
-            .remaining(self.order.data.buy_amount)
-            .context("remaining buy")?;
-
-        let sell_price = clearing_prices
-            .get(&self.order.data.sell_token)
-            .context("no clearing price for sell token")?;
-        let buy_price = self
-            .buy_token_price(clearing_prices)
-            .context("no clearing price for buy token")?;
-
-        let order = &self.order.data;
-        let (sell_amount, buy_amount, fee_amount) = match order.kind {
-            OrderKind::Sell => {
-                let sell_amount = self.executed_amount;
-                let buy_amount = sell_amount
-                    .checked_mul(*sell_price)
-                    .and_then(|v| v.checked_ceil_div(&buy_price))
-                    .context("could not compute buy amount")?;
-                let fee_amount = order
-                    .fee_amount
-                    .checked_mul(self.executed_amount)
-                    .and_then(|v| v.checked_div(order.sell_amount))
-                    .context("could not compute executed fee amount")?;
-                (sell_amount, buy_amount, fee_amount)
-            }
-            OrderKind::Buy => {
-                let buy_amount = self.executed_amount;
-                let sell_amount = buy_amount
-                    .checked_mul(buy_price)
-                    .and_then(|v| v.checked_div(*sell_price))
-                    .context("could not compute sell amount")?;
-                let fee_amount = order
-                    .fee_amount
-                    .checked_mul(self.executed_amount)
-                    .and_then(|v| v.checked_div(order.buy_amount))
-                    .context("could not compute executed fee amount")?;
-                (sell_amount, buy_amount, fee_amount)
-            }
-        };
-
-        let execution = TradeExecution {
-            sell_token: order.sell_token,
-            buy_token: order.buy_token,
-            sell_amount,
-            buy_amount,
-            fee_amount,
-        };
-
-        anyhow::ensure!(
-            execution.sell_amount <= remaining_sell && execution.buy_amount >= remaining_buy,
-            "limit prices not respected"
-        );
-
-        Ok(execution)
-    }
-}
-
-/// Contains all the information required to participate in the solver
-/// competition.
-#[derive(Debug, Default, Clone)]
-pub struct SettlementProposal {
-    pub clearing_prices: HashMap<H160, U256>,
-    pub trades: Vec<TradedOrder>,
-    pub execution_plan: Vec<Arc<dyn InteractionProposal>>,
-    pub auction_id: u64,
-}
-
-impl SettlementProposal {
```

Comment:
I think this still needs to be ported into the driver. We wanted the driver to verify solutions before submitting them to prevent accidental bad behavious from the `solver engine`s.